fusesoc/config.py

```diff
@@ -149,13 +162,14 @@ def add_library(self, name, library):
 
         self.libraries[name] = library
 
-        if not os.path.isdir(library['location']):
-            logger.info("Cloning library {}".format(name))
-            git_args = ['clone', library['sync-uri'], library['location']]
-            try:
-                Launcher('git', git_args).run()
-            except RuntimeError as e:
-                logger.error("`library add` failed: " + str(e))
+        try:
+            provider_module = importlib.import_module(
+                'fusesoc.provider.%s' % library['sync-type'])
+        except ImportError as e:
+            logger.error("Invalid sync-type '{}'".format(library['sync-type']))
+            exit(1)
```

Comment:
exit is only allowed in main.py. Reraise as RuntimeError("Invalid sync-type '{}'".format(library['sync-type'])) and catch it in main instead

Reply:
Done